resolve_stream.py

```diff
@@ -147,6 +261,57 @@ def resolve_stream(refdes, method, stream, stream_map, m2m):
                 print '%sNOT FOUND: needed_stream=%r poss_params=%r' % (indent, needed_stream, poss_params)
 
 
+# TODO - need to configure this to either use asset management directly or M2M
```

Comment:
Will a separate change follow up on the m2m config TODO? It looks like this same m2m config logic is present in the main method. Should m2m and stream_map be passed to fully_resolve_parameter as parameters to remove this redundancy?